submitter/submission.py

```diff
@@ -119,7 +117,7 @@ def add_bitstreams_to_item(self, item: dspace.item.Item) -> dspace.item.Item:
             logger.debug(
                 "Adding bitstreams to local item instance from submission message"
             )
-            for file in self.files:
+            for file in self.files or []:
```

Comment:
Could `self.files` default to an empty list or would that cause issues?

Reply:
The way the class is setup, `self.files` must be allowed to be `None` (in the case of an "invalid" message).  But, this bristles the linting because this is assuming -- if you've made it this far -- that `self.files` will be a list to iterate through and add bitstreams.  

This helps pass linting by _explicitly_ showing that if `self.files` still is `None` by this point, but we haven't bailed for an invalid message, we can act like it's an empty list and just not add any bitstreams.

I think this is kind of a nice example of `mypy` type linting.  The fact this looks a little awkward here I think actually reflects that it _is_ a little awkward to know if `self.files` will be hydrated by this point.  Much of the codebase assumes that properties are set on this `Submission` instance because the message was valid.  But many of those properties must be `None` by default to support the unhappy path.  

Nothing feels like a deal breaker, but my experience was that setting these type hints or small adjustments were revealing the underlying nature in a good way.

Reply:
👍